### ISSUE TITLE
Process never finished

### ISSUE THREAD

**Issue by cschan1828** [andyludeveloper/kotlin-coroutine-book-example#1]
Process never finished for this example, is it expected? I think the outer Coroutine should have not been exited before `println("done")`?
`src/main/kotlin/com/andyludeveloper/coroutine_book_example/ch5/Example11_Cancel_joinAll.kt`

**Comment by andyludeveloper** [comment]
Yes, you're right. the job.cancel() only cancel the job is started in the launch block.
So, after the job.cancel() the repeat block still generate the new job then these jobs will not be canceled.

**Comment by andyludeveloper** [comment]
In my book, i miss to describe the remain job still run, I'll added these information in the next version.
Thanks.
